### ISSUE TITLE
Can't marshal / serialize GoogleHash object

### ISSUE THREAD

**Issue by danyal** [rdp/google_hash#21]
Is there any way to marshall / searialze a google hash object? I would ideally like to create the hash, store it in a db, and then recreate it from the db.

**Comment by floere** [comment]
I'd be extremely interested in this, as I'd like to use this lib for my search engine, http://pickyrb.com/ – sorry that this is just a +1.

**Comment by rdp** [comment]
Could somebody please elaborate on what their use case is for this? They
need a super fast serial/deserialize to disk?

**Comment by floere** [comment]
In Picky, I'd like to be able to dump/load the Hash-based indexes to/from disk. Currently, I could use `#each` to dump to disk, and then load it again – however, if the faster version was available, I'd gladly use that.

**Comment by AlexRevin** [comment]
Yes, I am using it the same way: calculating millions of combinations and storing them in google_hash while in app. Cannot really save them anywhere because google_hash is not dumpable.
+1 a lot!

**Comment by rdp** [comment]
OK are you folks all just looking for dumping native types or Ruby objects?

**Comment by floere** [comment]
@rdp For me it's not as important how they are stored, just that they can be reloaded.

**Comment by rdp** [comment]
what type of objects, though, does this include ruby objects or just native
ones [storing ruby objects is far more tricky]

On Thu, Mar 19, 2015 at 10:25 AM, Florian R. Hanke <notifications@github.com

**Comment by AlexRevin** [comment]
I think storing native objects is enough. Storing custom objects requires marshaling which is not a good idea at all.

**Comment by floere** [comment]
@rdp By "native", do you mean String/Numeric/…?

**Comment by AlexRevin** [comment]
Exactly, + symbols

**Comment by floere** [comment]
@rdp For me, it would already be great if "native" objects would be supported.

**Comment by rdp** [comment]
Native means numeric in this case [I don't special store Strings yet,
though that would be interesting...]
